File: packages/api/aqapi/core/domain/value_object/base_id.py
from typing import Hashable, override
from aqapi.core.domain.value_object.base_value_object import BaseValueObject


class BaseId(BaseValueObject[int], Hashable):
    """ドメインモデルのIDを表す基底クラス"""

    def __init__(self, value: int):
        super().__init__(value)

    @override
    def _validate(self) -> None:
        self._validator.validate_required()
        self._validator.validate_integer()
        self._validator.validate_min_value(1)
    
    def __hash__(self) -> int:
        """ハッシュ値の計算（辞書のキーとして使用可能にする）"""
        return hash(self._value)
    
    def __int__(self) -> int:
        """IDを整数として返す"""
        return self._value
    

File: packages/flutter/lib/core/value_object/base_id.dart
import 'package:allowance_questboard/core/value_object/base_state_object.dart' show BaseValueObject;

/// ドメインモデルのIDを表す基底クラス
/// APサーバ側のBaseIdと同じ構造を持つ
abstract class BaseId extends BaseValueObject<int> {
  BaseId(super.value);
  
  @override
  void validate() {
    validator.required();
    validator.integer();
    validator.minValue(1);
  }
  
  /// IDを整数として返す
  int toInt() => value;

  @override
  String toString() => '$runtimeType($value)';
}
